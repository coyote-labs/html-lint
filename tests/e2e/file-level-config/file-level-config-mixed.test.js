const { getFileMeta } = require('../../utils.js');

global.console = {
  error: jest.fn().mockName('error'),
  warn: jest.fn().mockName('warn'),
  log: jest.fn().mockName('log')
};

describe('file-level-config', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test('mixed', async() => {
    expect.hasAssertions();

    try {
      const lint = require('../../../index');
      await lint([getFileMeta('e2e/file-level-config/fixtures/mixed.html')]);
    } catch ({ message }) {
      expect(global.console.error).toMatchSnapshot();
      expect(global.console.warn).toMatchSnapshot();
      expect(global.console.log).toMatchSnapshot();
    }
  });
});
