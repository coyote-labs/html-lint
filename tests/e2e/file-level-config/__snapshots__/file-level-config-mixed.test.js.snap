// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file-level-config mixed 1`] = `
[MockFunction error] {
  "calls": Array [
    Array [
      "[101m[html-lint] 1 error in e2e/file-level-config/fixtures/mixed.html[49m",
    ],
    Array [
      "[90mclass-ad-disabled: Class names should not contain \`ad(s)\` or \`sponsor(s)\` to avoid getting flagged by ad blockers.[39m
[0m [90m  5 | [39m  [33m<[39m[33m/[39m[33mhead[39m[33m>[39m[0m
[0m [90m  6 | [39m  [33m<[39m[33mbody[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m  7 | [39m    [33m<[39m[33mmain[39m [36mclass[39m[33m=[39m[32m\\"sponsors\\"[39m[33m>[39m[0m
[0m [90m    | [39m          [31m[1m^[22m[39m [31m[1mClass names should not contain \`ad(s)\` or \`sponsor(s)\` to avoid getting flagged by ad blockers.[22m[39m[0m
[0m [90m  8 | [39m      [33m<[39m[33mul[39m[33m>[39m[0m
[0m [90m  9 | [39m        [33m<[39m[33mli[39m id[33m=[39m[32m\\"company-ad\\"[39m[33m>[39m[33mCompany[39m [33mA[39m[33m<[39m[33m/[39m[33mli[39m[33m>[39m[0m
[0m [90m 10 | [39m        [33m<[39m[33mli[39m id[33m=[39m[32m\\"company-a\\"[39m[33m>[39m[33mCompany[39m [33mB[39m[33m<[39m[33m/[39m[33mli[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`file-level-config mixed 2`] = `
[MockFunction warn] {
  "calls": Array [
    Array [
      "[43m[30m[html-lint] 1 warning in e2e/file-level-config/fixtures/mixed.html[39m[49m",
    ],
    Array [
      "[90mdoctype-first: HTML files should begin with a \`doctype\`.[39m
[0m [90m 1 | [39m[33m<[39m[33m![39m[33m--[39m html[33m-[39mlint doctype[33m-[39mfirst[33m:[39mwarn[33m,[39m id[33m-[39mad[33m-[39mdisabled[33m:[39moff [33m--[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m 2 | [39m[33m<[39m[33mhtml[39m[33m>[39m[0m
[0m [90m   | [39m[31m[1m^[22m[39m [31m[1mHTML files should begin with a \`doctype\`.[22m[39m[0m
[0m [90m 3 | [39m  [33m<[39m[33mhead[39m[33m>[39m[0m
[0m [90m 4 | [39m    [33m<[39m[33mtitle[39m[33m>[39m[33mTest[39m[33m<[39m[33m/[39m[33mtitle[39m[33m>[39m[0m
[0m [90m 5 | [39m  [33m<[39m[33m/[39m[33mhead[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`file-level-config mixed 3`] = `[MockFunction log]`;
